### DISCUSSION TITLE
Is the Android / Amazon Mini Client still being supported?

### DISCUSSION THREAD

**Discussion by Sagetvmaster** [OpenSageTV/Community#44] [Android client]
Someone just replied that they didn't think Josh was still doing work or supporting the Mini-Client, anyone know if this is true?
There haven't been any entries here in a couple of weeks, or I don't know where to find them using this new site.
Anyone know what's going on?

**Comment by techiegirl2day** [comment]
I would like to know as well.  With the changes to Firesticks ( not for the better ) I am contemplating my future configuration for hardware/ software combinations.  With the Media Content on "pause" due to strike in Hollywood it is making me take a hard look at my entertainment options, streaming vs recording shows etc.  I dont know how much longer we will be able to use our dvr setups if they start encrypting the signals. How many developers do we still have onboard with this project? This feels like we are going through a slow inevitable death, with pvrs becoming another relic of the past.  Thoughts? 

Galaxysurfer

**Reply by nyplayer** [reply]
SageTV is just too behind in the times there is no IPTV Support nor M3U support... there are just better options out there. The only thing that has kept SageTV on life support in my household is using NEXTPVR as a backend.

**Reply by graywolfusmc** [reply]
So you are using NextPVR to handle the scheduling but playback with SageTV?
I assume that the HD200/300s would not work with NextPVE, how do you handle setting recordings in NextPVR? What do you use?

**Reply by nyplayer** [reply]
No I have NEXTPVR handle the sources and feed M3U/XMLTV into OPENDCT use SageTV to record and Schedule.

**Comment by techiegirl2day** [comment]
What do you suggest for the Sagetv dvr/player alternatives ( it is my main use of Sagetv) ? I dont want to pay for all those cloud media streaming services at this point.  Keep in mind I am in Canada so Channels dvr not supported here. Is Nextpvr my main option?

**Reply by Sagetvmaster** [reply]
I’ve found Jellyfin, Plex & Air Video HD all competent ways to play back recorded material, but still, for my uses (aka I don’t need 4k) SageTV is still the most flexible recording DVR when coupled with appropriate tuners.  But for cable & satellite, not so much streaming.  What’s your primary use?

**Comment by techiegirl2day** [comment]
Basically, at this point it is my home dvr setup interfacing with HDHomerun for OTA recording.  I also use it as a way to connect all TVs to antenna without having to rewire house or install antenna differently. ( I am not overly fond of HDHomerun app) I prefer my onsite recording to access content rather than from the cloud aka stream ( which they only give you a week to access after original airing if at all ) except as backup when dvr fails to record, making it less necessary to need internet other than for guide updates or getting Epg directly from tv stations. 
     I know inevitably the days of free TV are going to end soon if the vested interests have their way as opposed to old model of ads to pay for it. I am not willing to pay for streamed tv that still has mandatory ads along with paid subscription. Talk about Double dipping! Guess at that time will go old school & buy books for entertainment etc. esp if move away to country where all that is less easy to access.  The control & greed is never ending.

**Reply by Sagetvmaster** [reply]
Well, let's hope the FCC remembers why they were set up in the first place: to serve the public and protect their viewing rights.  OTA is still one of the best ways to watch TV if you're within range.  Sadly for me, we're exactly 15 miles out of range between the two, biggest, nearest metropolis's.  If we were within the boundaries of one or the other, I'd be paying for that HDHomerun DVR service for sure, as well as using Sage.

**Comment by eelpout** [comment]
I don't think the [source code](https://github.com/OpenSageTV/sagetv-miniclient) has been touched since last year. Josh disappeared a while ago back on the old forum.

It would be nice to at least have the ExoPlayer, IJKPlayer and ffmpeg parts updated in the client every once in a while. And to make closed captioning actually work. :)

**Reply by Sagetvmaster** [reply]
Did he disappear?  I would understand if his Patreon $$ wasn’t enough, but I thought he was a pretty big fan of SageTV and using the miniclient, hence a personal hobby.  Maybe he left Sage behind?

**Reply by eelpout** [reply]
_Did he disappear?_ 
as far as I can tell, yes.

he was a big fan, yep. in addition to the miniclient he also was doing work on a new transcoder and the mediaformat plugin. he had stopped posting months ago on the old forums. no pre-notice for an exit as far as I know.

unfortunately, the old forums are a mess right now and can't even be searched to find his last post. not sure what's happening there.

**Comment by TGsell1** [comment]
Closed captioning wouldBe extremely valuable! Josh did a lot of excellent work as recently as the end of last year and I believe he’s still active as I support his Patreon site.  Sent from my iPhoneOn Jul 17, 2023, at 7:29 PM, eelpout ***@***.***> wrote:﻿
I don't think the source code has been touched since last year. Josh disappeared a while ago back on the old forum.
It would be nice to at least have the ExoPlayer, IJKPlayer and ffmpeg parts updated in the client every once in a while. And to make closed captioning actually work. :)

**Reply by eelpout** [reply]
have you tried to contact him by chance to see what's up?

**Reply by nyplayer** [reply]
I was a Patreon supporter and left because I had not heard from him in a while and still have not heard from him.

**Comment by wayner9** [comment]
As long as you don't care about 4K or streaming support on the SageTV device then just buy some used HD300s off of someone.  I have dabbled with the Android TV client at times but HD300s seems way more stable, especially for OTA recrodings.

**Comment by techiegirl2day** [comment]
Anyone here a subscriber to Josh's Patreon channel?
 https://www.patreon.com/jvl711 

I just checked his account & noticed he made a new post today or yesterday.  I am not a Patreon subscriber so I cant communicate with him there.   Can someone get word from him to pop into the community channel here?



Galaxysurfer

**Reply by way-lo** [reply]
I do contribute to his Patreon but apparently not enough to meet the 'member' criteria.  There was another update 2 weeks ago.  But still blocked for non-members.

I still use the Android client quite frequently on my Shield devices.  Works just fine.

EDIT: Oops, I'm actually a Patreon for Ken Birch, not Josh.

**Reply by eelpout** [reply]
he hasn't made any commits in any of his SageTV projects recently that I've seen so not sure what he's working on now.

**Reply by LoisGNS** [reply]
One of the older versions - 1.28 I think, works on my tablets, but the newer ones don't. The newer ones don't even start up properly.

**Comment by nyplayer** [comment]
If you are having Problems Playing recorded Content ... this is what I do I make sure I have the latest SageTVTranscoder Plugin ... I have all my favorites set to Convert to MKV automatically. Using my Property settings below ... I use the Remux one. Then on my FireTV devices I have use EXOPLAYER with FFMPEG Audio Extension. No Problems Seeking or Playing.

Transcode MP4

transcoder/formats/mp4-H264fast\ mp4=f\=mp4;MCompressionDetails\=-preset fast -c\:v libx264;
transcoder/formats/mp4-H264faster\ mp4=f\=mp4;MCompressionDetails\=-preset faster -c\:v libx264;
transcoder/formats/mp4-H264Medium\ mp4=f\=mp4;MCompressionDetails\=-preset medium -c\:v libx264;
transcoder/formats/mp4-H264slow\ mp4=f\=mp4;MCompressionDetails\=-preset slow -c\:v libx264;
transcoder/formats/mp4-H264slower\ mp4=f\=mp4;MCompressionDetails\=-preset slower -c\:v libx264;
transcoder/formats/mp4-H264superfast\ mp4=f\=mp4;MCompressionDetails\=-preset superfast -c\:v libx264;
transcoder/formats/mp4-H264veryfast\ mp4=f\=mp4;MCompressionDetails\=-preset veryfast -c\:v libx264;
transcoder/formats/mp4-H264veryslow\ mp4=f\=mp4;MCompressionDetails\=-preset veryslow -c\:v libx264;


Transcode MKV keep audio.

transcoder/formats/MKV-H264fast\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset fast -c\:v libx264;
transcoder/formats/MKV-H264faster\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset faster -c\:v libx264;
transcoder/formats/MKV-H264Medium\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset medium -c\:v libx264;
transcoder/formats/MKV-H264slow\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset slow -c\:v libx264;]
transcoder/formats/MKV-H264slower\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset slower -c\:v libx264;
transcoder/formats/MKV-H264superfast\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset superfast -c\:v libx264;
transcoder/formats/MKV-H264veryfast\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset veryfast -c\:v libx264;
transcoder/formats/MKV-H264veryslow\ MKV=f\=matroska;MCompressionDetails\=-c copy -preset veryslow -c\:v libx264;
transcoder/formats/MKV-Remux\ MKV=f\=matroska;MCompressionDetails\=-c copy;

![image](https://github.com/OpenSageTV/Community/assets/16063354/76c978df-9b22-4284-8a61-ca26999fb000)

![image](https://github.com/OpenSageTV/Community/assets/16063354/987c5ab3-4240-4993-8dda-60f781e9bb97)

**Reply by nyplayer** [reply]
Remux does not transcode it simply converts them to MKV and fixes any errors it might have.  But you can use the other MKV settings to transcode if you want.

![image](https://github.com/OpenSageTV/Community/assets/16063354/e6ac340d-a862-473d-8921-bcde816ea9ed)

**Reply by eelpout** [reply]
hey nyplayer, if I may ask, what is your current server hardware & OS software? 

I long ago gave up on using SageTV's built-in conversion because I found recording (I have up to 7 tuners) and converting simultaneously was overwhelming my own setup. So, I punted my MKV conversions to overnight using SJQ and MediaShrink. I will admit that my AMD/Windows setup is getting a little aged. ;)

**Reply by nyplayer** [reply]
The built in SageTV Transcoder does them one at a time and runs in low Priority so it should not overwhelm your system.

**Reply by eelpout** [reply]
agree, it _shouldn't_, but whenever I've done this in the past, it affects any playback that may to be going on from any file with stuttering. often there are other processes going on too like Comskip etc. 

though I haven't tried this with jv711's version of the transcoder so maybe I'll give it a whirl.

**Reply by LoisGNS** [reply]
Where do you enter these additional transcode settings? I am using the built-in version for conversions, and its settings are in sage.properties, but I'm not sure how to fit mkv into the required format there (which I don't really understand).

**Reply by nyplayer** [reply]
The built in Version no longer work you have to shutdown SageTV and copy them to Sage.properties.

**Reply by LoisGNS** [reply]
I use the built-in conversion all the time, but just need the right parameters to add to .properties to add additional formats. For what actions does the built-in no longer work? Maybe we're talking about different things?

**Reply by nyplayer** [reply]
I am tired of trying to get people to update when they are not willing to try so I am outta this forum good luck all. You can lead a horse to water but you cannot make him drink 

Bye

**Reply by Sagetvmaster** [reply]
Hey nyplayer - I’m up to date, at least I think I am.  What is this you’re referring to?

**Reply by eelpout** [reply]
ah, just _grokked_ you are using remux, which uses basically no resources. :) 

since I like to shrink mine at the same time, I'll keep using my overnight MKV method. unless I do a deep pruning of old shows to free up space, which I should really do.

**Reply by trk2** [reply]
The transcoder plugin also supports hardware encoding.  If you have a decent video card, you can also greatly reduce cpu resources, by putting the video card to work.  Also, you can configure the miniclient to use a fixed transcode or remux format during playback.  I use the miniclient every day to flawlessly play live and recorded tv, as well as 4k, HDR hevc content.  But, I agree with nyplayer, it requires keeping up to date on plugins (mainly the transcoder plugin and format detector plugin) and to have a relatively modern PC.  You can't expect to use stock Sage with a 15 year old version of ffmpeg on a 15 year old PC and have good performance (or even work) with modern codecs.

Also, you have to move away from mpeg2 which has poor support for Android.  It's an ancient codec anyway, and there's not really an excuse to not be using h.264.

**Reply by nyplayer** [reply]
Exactly Well Said.

**Reply by eelpout** [reply]
> Also, you have to move away from mpeg2 which has poor support for Android. It's an ancient codec anyway, and there's not really an excuse to not be using h.264.

that may be, but those of us with sources of a certain age, i.e., Silicon Dust Primes, H. HD-PVR's, have MPEG-2 transport streams natively. but those of course can be converted.

**Reply by trk2** [reply]
Depending on the Android device, you may have good support for mpeg2 and this is not necessary.  I use a Prime.  Hopefully your service is like mine and the only mpeg2 channels are the local broadcast networks.  But there are a many possible solutions here.  OpenDCT can be configured to transcode only mpeg2 streams during capture. The miniclient can be configured to transcode mpeg2 streams on playback.  Or you can convert within Sage itself, either manually or automatically.  My point is that Sage with plugins has the features if you have sufficient hardware to take advantage of it.

**Comment by nyplayer** [comment]
Time for the users that have not touched Sage in years to get up to date.
